libs/designer/src/lib/ui/panel/configureTemplatePanel/configureWorkflowsPanel/usePanelTabs.tsx

```diff
@@ -2,33 +2,86 @@ import type { TemplateTabProps } from '@microsoft/designer-ui';
 import { selectWorkflowsTab } from './tabs/selectWorkflowsTab';
 import { customizeWorkflowsTab } from './tabs/customizeWorkflowsTab';
 import { useIntl } from 'react-intl';
-import { useDispatch } from 'react-redux';
-import type { AppDispatch } from '../../../../core/state/templates/store';
+import { useDispatch, useSelector } from 'react-redux';
+import type { AppDispatch, RootState } from '../../../../core/state/templates/store';
+import { useFunctionalState } from '@react-hookz/web';
+import type { WorkflowTemplateData } from '../../../../core';
+import { updateAllWorkflowsData } from '../../../../core/state/templates/templateSlice';
+import { getWorkflowsWithDefinitions, initializeWorkflowsData } from '../../../../core/actions/bjsworkflow/configuretemplate';
+import { getResourceNameFromId } from '@microsoft/logic-apps-shared';
 
 export const useConfigureWorkflowPanelTabs = ({
-  onClosePanel,
   onSave,
 }: {
-  onClosePanel: () => void;
   onSave?: (isMultiWorkflow: boolean) => void;
 }): TemplateTabProps[] => {
   const intl = useIntl();
   const dispatch = useDispatch<AppDispatch>();
+  const { workflowsInTemplate, workflowState } = useSelector((state: RootState) => ({
+    workflowsInTemplate: state.template.workflows,
+    workflowState: state.workflow,
+  }));
 
   const hasError = false; // Placeholder for actual error state
   const isSaving = false; // Placeholder for actual saving state
 
+  const [selectedWorkflowsList, setSelectedWorkflowsList] =
+    useFunctionalState<Record<string, Partial<WorkflowTemplateData>>>(workflowsInTemplate);
+
+  const onWorkflowsSelected = (normalizedWorkflowIds: string[]) => {
+    setSelectedWorkflowsList((prevSelectedWorkflows) => {
+      const newSelectedWorkflows: Record<string, Partial<WorkflowTemplateData>> = {};
+      for (const normalizedWorkflowId of normalizedWorkflowIds) {
+        newSelectedWorkflows[normalizedWorkflowId] = prevSelectedWorkflows[normalizedWorkflowId] ?? {
+          id: normalizedWorkflowId,
+          workflowName: getResourceNameFromId(normalizedWorkflowId),
+        };
+      }
+      return newSelectedWorkflows;
+    });
+  };
+
+  const updateWorkflowDataField = (workflowId: string, workflowData: Partial<WorkflowTemplateData>) => {
+    setSelectedWorkflowsList((prevSelectedWorkflows) => ({
+      ...prevSelectedWorkflows,
+      [workflowId]: {
+        ...prevSelectedWorkflows[workflowId],
+        ...workflowData,
+      },
+    }));
+  };
+
+  const onNextButtonClick = async () => {
+    setSelectedWorkflowsList(await getWorkflowsWithDefinitions(workflowState, selectedWorkflowsList()));
+  };
+
+  const onSaveChanges = (isMultiWorkflow: boolean) => {
+    onSave?.(isMultiWorkflow);
```

Comment:
I think, you should remove the isMultiWorkflow from here, and call onSave?.(workflowsList().length > 1) from here.
move this call after the last dispatch in this method